### PR TITLE
fix: Fn+F fullscreen transitioning on macOS

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Closes https://github.com/electron/electron/issues/37761.

Fixes an issue on macOS where entering fullscreen with the `Fn+F` system shortcut would fail or create strange window side effects. This was happening as a result of https://chromium-review.googlesource.com/c/chromium/src/+/3496322, which changed the logic in `content/app_shim_remote_cocoa/render_widget_host_view_cocoa.mm` to sent the keyboard shortcut directly to the `NSMenu` instance if the keyboard event is a system shortcut. In combination with our own logic in [`WebContents::PlatformHandleKeyboardEvent`](https://github.com/electron/electron/blob/5d120359f6f6adf730455189d099b0a63a0e5cc9/shell/browser/api/electron_api_web_contents_mac.mm#L54-L57), this meant that the BrowserWindow would be incorrectly sent _two_ calls to `toggleFullScreenMode`. We prevent this by ensuring that system keyboard shortcuts are only sent to the `NSMenu` instance once.

#### Checklist

- [ ] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [ ] relevant documentation, tutorials, templates and examples are changed or added
- [ ] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: Fixes an issue on macOS where entering fullscreen with the `Fn+F` system shortcut would fail or create strange window side effects.

### CODE REVIEW

File: shell/browser/api/electron_api_web_contents_mac.mm [hunk 1]
```diff
@@ -52,9 +53,20 @@ - (void)redispatchKeyEvent:(NSEvent*)event;
...
-    return true;
+  if (event.os_event.type == NSEventTypeKeyDown) {
+    // If the keyboard event is a system shortcut, it's already sent to the
+    // NSMenu instance in
+    // content/app_shim_remote_cocoa/render_widget_host_view_cocoa.mm via
+    // keyEvent:(NSEvent*)theEvent wasKeyEquivalent:(BOOL)equiv. If we let the
+    // NSMenu handle it here as well, it'll be sent twice with unexpected side
+    // effects.
+    auto modifier_mask = ui::cocoa::ModifierMaskForKeyEvent(event.os_event);
+    if ((modifier_mask & NSEventModifierFlagFunction) != 0)
```
Comment (zcbenz) [inline shell/browser/api/electron_api_web_contents_mac.mm:64]: I think we should also do a test of `[[NSApp mainMenu] cr_menuItemForKeyEquivalentEvent:theEvent] != nil` like the original Chromium change did, otherwise other valid Fn key events won't be sent to renderer.

Comment (codebytere) [reply shell/browser/api/electron_api_web_contents_mac.mm:64]: @zcbenz done!

**Review by zcbenz — APPROVED** [review]
